exercises/perfect-numbers/perfect_numbers_test.go

```diff
@@ -0,0 +1,39 @@
+package perfect
+
+import "testing"
+
+func TestClassifiesCorrectly(t *testing.T) {
+	cases := []struct {
+		input    uint64
+		expected Classification
+	}{
+		{1, ClassificationDeficient},
+		{13, ClassificationDeficient},
+		{12, ClassificationAbundant},
+		{6, ClassificationPerfect},
+		{28, ClassificationPerfect},
+		{496, ClassificationPerfect},
+		{8128, ClassificationPerfect},
+	}
+	for _, c := range cases {
+		if cat, err := Classify(c.input); err != nil {
+			t.Errorf("%d: Expected no error but got %s", c.input, err)
+		} else if cat != c.expected {
+			t.Errorf("%d: Expected %q, got %q", c.input, c.expected, cat)
+		}
+	}
+}
+
+func TestGivesPositiveRequiredError(t *testing.T) {
+	if _, err := Classify(0); err != ErrOnlyPositive {
+		t.Errorf("Expected eror %q but got %q", ErrOnlyPositive, err)
```

Comment:
Do you think that we need to do the "ensure the `err` return is actually an `error`" of https://github.com/exercism/xgo/issues/275, or would the name `ErrOnlyPositive` starting with `Err` convince the student enough? The goal is simply to encourage idiomatic Go code, so if thinking the name is good enough, we don't need to add an explicit compile-time check.